### PR TITLE
GL-1101 optional sample name and sample id from user to loom

### PR DESCRIPTION
**Pull request by khajoue2**
Inputs to the Optimus and SS2 pipelines:

- `input_id` - the value of `sequencing_process.provenance.document_id`, which is a UUID. Previously called sample_id in the pipeline. Unique across the HCA and used to name bam and loom files.
- 
- input_id_metadata_field [optional] - will be the name of the metadata field used for the input_id, in this case it will be `sequencing_process.provenance.document_id`. Will appear in loom file as a global attribute (10X) or per cell (SS2) as  sequencing_process.provenance.document_id: [UUID] .
- 
- `input_name` [optional]  - human readable version of input_id to hopefully connect DCP matrices to author matrices/annotations. Will be pulled from sequencing_input.biomaterial_core.biomaterial_id in HCA metadata. Will be a global attribute in loom file (10X) or per cell (SS2).
- 
- `input_name_metadata_field` [optional] - will be the name of the metadata field used for input_name, in this case it will be `sequencing_input.biomaterial_core.biomaterial_id`. Will appear in loom file as a global attribute (10X) or per cell (SS2) as  `sequencing_input.biomaterial_core.biomaterial_id`: [name].

- Merged terra compatibility branch into this branch with checker wdls

-Updated and activated ss2 pipeline tests

### CODE REVIEW

File: pipelines/skylab/smartseq2_multisample/Loom_schema.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ The global attributes in the Loom apply to the whole file, not any specific part
 | :-- | :-- |
 | LOOM_SPEC_VERSION | String with the loom file spec version |
 | CreationDate | Date Loom file was generated |
-| sample_id | The plateid listed in the pipeline configuration file; by default it is set to the batch_id. |
+| sample_id | The plateid listed in the pipeline configuration file; by default it is set to the batch_id |
```
Comment (ekiernan) [inline pipelines/skylab/smartseq2_multisample/Loom_schema.md:14]: Does the global attribute still say sample_id on the Multi Sample SS2?

Comment (khajoue2) [reply pipelines/skylab/smartseq2_multisample/Loom_schema.md:14]: It is batch_id and batch_name now

Comment (kbergin) [reply pipelines/skylab/smartseq2_multisample/Loom_schema.md:14]:
```suggestion
| sequencing_input_id | The plate_id listed in the pipeline configuration file; by default it is set to the batch_id |
```

**Comment by kbergin** [comment]
@ekiernan I added the documentation folder when bringing over the pipelines because optimus in WARP already had that structure so I thought that was the style. Sorry for the confusion! In the Optimus one, the readme was still in the main Optimus folder but the documentation folder had all the other png images and content. Like we talked about on slack we can decide on an organization of the whole repo after everything is moved over

**Comment by kbergin** [comment]
@khajoue2 let me know when you have the latest changes needed to the naming in this PR and I'll review. Jess or another eng should as well I imagine.

**Comment by pshapiro4broad** [comment]
Make sure you remove the `.DS_Store ` files from this PR, since they are mac-specific desktop files that shouldn't be committed to git.

File: pipelines/skylab/optimus/Optimus.changelog.md [hunk 2]
```diff
@@ -1,10 +1,22 @@
+# optimus_v4.0.0
+
+2020-08-10 (Date of Last Commit)
+### Breaking changes to the pipeline inputs.
```
Comment (ekiernan) [inline pipelines/skylab/optimus/Optimus.changelog.md:4]: Do we want to demarcate breaking changes moving forward for all pipelines? If so, I can add to the changelog style guide

Comment (khajoue2) [reply pipelines/skylab/optimus/Optimus.changelog.md:4]: Yes, I think it is a good idea!

**Pull request merged**